carbon-design-system/carbon
# feat(search): add support for expandable variant

Closes #6360

Adds an expandable variant to the search component.

#### Changelog

**New**

- Expandable variant / kind for search component

**Changed**

- Renamed previous `bx--search-magnifier` classname to `bx--search-magnifier-icon`
  - If changing classnames is considered a breaking change, I can also revert this and change the new `bx--search-magnifier` to `bx--search-magnifier-container` – just let me know!
- Updated API snapshot to reflect new prop

#### Testing / Reviewing

When `props.kind = 'expandable'`, the search component should
- be collapsed by default
- open by either
  - clicking the search icon
  - tabbing into it / receiving focus
- close when it loses focus and has no content
- stay open if it loses focus and has content
